src/Service.php

```diff
@@ -45,4 +46,49 @@ protected function getData(string $method, string $url, array $options = [])
             return [];
         }
     }
+
+    /**
+     * Build XML metadata element from metadata name-value pairs.
+     *
+     * Metadata pairs will be set as attributes on a <z:row xmlns:z='#RowsetSchema'/> element, e.g.
+     *
+     *   ['ows_CustomProperty' => 'Another prop value', 'ows_CCMMustBeOnPostList' => 0]
+     *
+     * will be converted to
+     *
+     *   <z:row xmlns:z="#RowsetSchema" ows_CustomProperty="Another prop value" ows_CCMMustBeOnPostList="0"/>
+     */
+    public function buildMetadata(array $metadata): string
```

Comment:
Why is this made public? Shouldn't it be kept internal, i.e. protected, in this class and subclasses? Callers should just pass metadata as arrays to the services.

Reply:
Good point. Updated `createCase` method to accept (and handle) metadata as an array.